ibm/service/vmware/resource_ibm_vmaas_transit_gateway_connection.go

```diff
@@ -0,0 +1,545 @@
+			MaxValueLength:             128,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "accept_language",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^[A-Za-z0-9-,;=\.\*\s]{1,256}$`,
+			MinValueLength:             1,
+			MaxValueLength:             256,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "region",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^.+$`,
+			MinValueLength:             1,
+			MaxValueLength:             25,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "vmaas_transit_gateway_connection_id",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^[A-Za-z0-9_-]{1,128}$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_vmaas_transit_gateway_connection", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIbmVmaasTransitGatewayConnectionCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vmwareClient, err := meta.(conns.ClientSession).VmwareV1()
+	if err != nil {
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_vmaas_transit_gateway_connection", "create", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	addTransitGatewayConnectionsOptions := &vmwarev1.AddTransitGatewayConnectionsOptions{}
+
+	addTransitGatewayConnectionsOptions.SetVdcID(d.Get("vdc_id").(string))
+	addTransitGatewayConnectionsOptions.SetEdgeID(d.Get("edge_id").(string))
+	addTransitGatewayConnectionsOptions.SetID(d.Get("vmaas_transit_gateway_connection_id").(string))
+
+	region := ""
+	if v, ok := d.GetOk("region"); ok {
+		region = v.(string)
+		addTransitGatewayConnectionsOptions.SetRegion(d.Get("region").(string))
+	}
+	acceptLang := ""
+	if v, ok := d.GetOk("accept_language"); ok {
+		acceptLang = v.(string)
+		addTransitGatewayConnectionsOptions.SetAcceptLanguage(d.Get("accept_language").(string))
+	}
+
+	transitGateway, _, err := vmwareClient.AddTransitGatewayConnectionsWithContext(context, addTransitGatewayConnectionsOptions)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("AddTransitGatewayConnectionsWithContext failed: %s", err.Error()), "ibm_vmaas_transit_gateway_connection", "create")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	d.SetId(fmt.Sprintf("%s/%s/%s/%s/%s", *addTransitGatewayConnectionsOptions.VdcID, *addTransitGatewayConnectionsOptions.EdgeID, *transitGateway.ID, acceptLang, region))
```

Comment:
why we need to set acceptlang and regjon part of ID

Reply:
both are optional as well